README.md
# Easy Flash Messages

![Example of Error Notification](https://i.imgur.com/6UnNsnp.png)

# Copyright
Inspired by [DevPromtheus98 Flash Package](https://github.com/DevPromtheus98/flashy).

## Installation

### You like text ?

First, pull in the package through Composer.

Run `composer require dev-promtheus98/flashy`

Note that the package are been tests only on laravel 9

## Usage

Within your controllers, before you perform a redirect...

```php
public function store()
{
    Flashy::message('Welcome Aboard!', 'http://your-awesome-link.com');

    return Redirect::home();
}
```

You may also do:

- `Flashy::info('Message', 'http://your-awesome-link.com')`
- `Flashy::success('Message', 'http://your-awesome-link.com')`
- `Flashy::error('Message', 'http://your-awesome-link.com')`
- `Flashy::warning('Message', 'http://your-awesome-link.com')`
- `Flashy::primary('Message', 'http://your-awesome-link.com')`
- `Flashy::primaryDark('Message', 'http://your-awesome-link.com')`
- `Flashy::muted('Message', 'http://your-awesome-link.com')`
- `Flashy::mutedDark('Message', 'http://your-awesome-link.com')`

Again, if using Laravel, this will set a few keys in the session:

- 'flashy_notification.message' - The message you're flashing
- 'flashy_notification.type' - A string that represents the type of notification (good for applying HTML class names)
- 'flashy_notification.link' - The URL to redirect to on click

Alternatively, again, if you're using Laravel, you may reference the `flashy()` helper function, instead of the facade. Here's an example:

```php
/**
 * Destroy the user's session (logout).
 *
 * @return Response
 */
public function destroy()
{
    Auth::logout();

    flashy()->success('You have been logged out.', 'http://your-awesome-link.com');

    return home();
}
```

Or, for a general information flash, just do: `flashy('Some message', 'http://your-awesome-link.com');`.

With this message flashed to the session, you may now display it in your view(s). Maybe something like:

```html
@if(Session::has('flashy_notification.message'))
<script id="flashy-template" type="text/template">
    <div class="flashy flashy--{{ Session::get('flashy_notification.type') }}">
        <i class="material-icons">speaker_notes</i>
        <a href="#" class="flashy__body" target="_blank"></a>
    </div>
</script>

<script>
    flashy("{{ Session::get('flashy_notification.message') }}", "{{ Session::get('flashy_notification.link') }}");
</script>
@endif
```

Because flash messages are so common, if you want, you may use (or modify) the views that are included with this package. Simply append to your layout view:

```html
@include('flashy::message')
```

> Note that this package has jQuery has dependency. It's also better to load flashy before your body close tag.

## Example

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Document</title>
</head>
<body>

<div class="container">

    <p>Welcome to my website...</p>
</div>

<script src="//code.jquery.com/jquery.js"></script>
@include('flashy::message')
</body>
</html>
```

If you need to modify the flash message partials, you can run:

```bash
php artisan vendor:publish
```

The two package views will now be located in the `app/views/packages/dev-promtheus98/flashy/` directory.

```php
Flashy::message('Welcome aboard!', 'http://your-awesome-link.com');

return Redirect::home();
```

```php
Flashy::error('Sorry! Please try again.', 'http://your-awesome-link.com');

return Redirect::home();
```

## Nice rendering

For a nice rendering you may include these lines in your head:

```html
<link href="//fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
<link href='//fonts.googleapis.com/css?family=Source+Sans+Pro:300,400,600,700' rel='stylesheet'>
```

and override the following sections of the default flashy view:

```html
<style type="text/css">
.flashy {
    font-family: "Source Sans Pro", Arial, sans-serif;
    padding: 11px 30px;
    border-radius: 4px;
    font-weight: 400;
    position: fixed;
    bottom: 20px;
    right: 20px;
    font-size: 16px;
    color: #fff;
}
</style>

<script id="flashy-template" type="text/template">
    <div class="flashy flashy--{{ Session::get('flashy_notification.type') }}">
        <i class="material-icons">speaker_notes</i>
        <a href="#" class="flashy__body" target="_blank"></a>
    </div>
</script>
```


CHANGELOG.md
# Changelog

All notable changes to `flashy` will be documented in this file.


src/views/message.blade.php
<style>
    .flashy {
        font-family: Arial, sans-serif;
        padding: 11px 30px;
        border-radius: 4px;
        font-weight: 400;
        position: fixed;
        bottom: 20px;
        right: 20px;
        font-size: 16px;
        color: #fff;
    }
    .flashy--success {
        background-color: #99c93d;
        color: #fff;
    }
    .flashy--warning {
        color: #8a6d3b;
        background-color: #fcf8e3;
        border-color: #faebcc;
    }
    .flashy--muted {
        background: #eee;
        color: #3a3a3a;
        border: 1px solid #e2e2e2;
    }
    .flashy--muted-dark {
        background: #133259;
        color: #e2e1e7;
    }
    .flashy--info a,
    .flashy--primary-dark a {
        color: #fff;
    }
    .flashy--error {
        background: #d14130;
        color: #fff;
    }
    .flashy--primary {
        background: #573e81;
    }
    .flashy--primary-dark {
        background: linear-gradient(to right, #133259 30%, #0d233e);
    }
    .flashy--info {
        background: #00baf3;
    }
    .flashy > ul {
        padding-left: 15px;
    }
    .flashy > p:only-of-type {
        margin-bottom: 0;
    }
    .flashy i {
        margin-right: 8px;
        position: relative;
        top: 6px;
    }
    .flashy .flashy__body {
        color: inherit;
    }
    @media only screen and (max-width:1050px) {
        .flashy {
            text-align: center;
            right: 0;
            left: 50%;
            width: 300px;
            margin-left: -150px;
        }
    }
    </style>
    
    <script>
        function flashy(message, link) {
            var template = $($("#flashy-template").html());
            $(".flashy").remove();
            template.find(".flashy__body").html(message).attr("href", link || "#").end()
             .appendTo("body").hide().fadeIn(300).delay(2800).animate({
                marginRight: "-100%"
            }, 300, "swing", function() {
                $(this).remove();
            });
        }
    </script>
    
    @if(Session::has('flashy_notification.message'))
    <script id="flashy-template" type="text/template">
        <div class="flashy flashy--{{ Session::get('flashy_notification.type') }}">
            <a href="#" class="flashy__body" target="_blank"></a>
        </div>
    </script>
    
    <script>
        flashy("{{ Session::get('flashy_notification.message') }}", "{{ Session::get('flashy_notification.link') }}");
    </script>
    @endif